### ISSUE TITLE
Errors: Can't conform to ImagePickerDelegate and Cannot assign self to image picker controller delegate

### ISSUE THREAD

**Issue by pjr0009** [hyperoslo/ImagePicker#236]
I'm using the basic example code, as well as trying using the code mentioned in #164 to fix the issue. Here are the errors. followed by the example code:
```1: Type 'ViewController' does not conform to protocol 'ImagePickerDelegate'```

```
Additional Info:
ImagePicker.ImagePickerDelegate:2:17: Protocol requires function 'wrapperDidPress(_:images:)' with type '(ImagePickerController, [UIImage]) -> ()'; do you want to add a stub?
ViewController.swift:26:10: Candidate has non-matching type '(ImagePickerController, [UIImage]) -> ()'
ImagePicker.ImagePickerDelegate:2:17: Protocol requires function 'wrapperDidPress(_:images:)' with type '(ImagePickerController, [UIImage]) -> ()'; do you want to add a stub?
ViewController.swift:26:10: Candidate has non-matching type '(ImagePickerController, [UIImage]) -> ()'
ImagePicker.ImagePickerDelegate:3:17: Protocol requires function 'doneButtonDidPress(_:images:)' with type '(ImagePickerController, [UIImage]) -> ()'; do you want to add a stub?
ViewController.swift:29:10: Candidate has non-matching type '(ImagePickerController, [UIImage]) -> ()'
ImagePicker.ImagePickerDelegate:3:17: Protocol requires function 'doneButtonDidPress(_:images:)' with type '(ImagePickerController, [UIImage]) -> ()'; do you want to add a stub?
ViewController.swift:29:10: Candidate has non-matching type '(ImagePickerController, [UIImage]) -> ()'
ImagePicker.ImagePickerDelegate:4:17: Protocol requires function 'cancelButtonDidPress' with type '(ImagePickerController) -> ()'; do you want to add a stub?
ViewController.swift:32:10: Candidate has non-matching type '(ImagePickerController) -> ()'
ImagePicker.ImagePickerDelegate:4:17: Protocol requires function 'cancelButtonDidPress' with type '(ImagePickerController) -> ()'; do you want to add a stub?
NewListingViewController.swift:32:10: Candidate has non-matching type '(ImagePickerController) -> ()'
```
note that adding a stub via xcode's fix-it feature does not fix the issue

``` 
2: ViewController.swift:18:42: Cannot assign value of type 'ViewController' to type '(UIImagePickerControllerDelegate & UINavigationControllerDelegate)?'
```

``` 
import UIKit
import ImagePicker

class ViewController: UIViewController {
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        let imagePickerController = ImagePickerController()
        imagePickerController.delegate = self
        
        //Present ImagePickerController
    }
}

extension ViewController: ImagePickerDelegate {
    
    func wrapperDidPress(_ imagePicker: ImagePickerController, images: [UIImage]) {
    }
    
    func doneButtonDidPress(_ imagePicker: ImagePickerController, images: [UIImage]) {
    }
    
    func cancelButtonDidPress(_ imagePicker: ImagePickerController) {
    }
}

```

Also, without conforming to the delegate, I only see the image library and no option to take a photo. I suspect I'll need to conform my controller to the delegate to use the camera?

**Comment by onmyway133** [comment]
@pjr0009 Hi, just want to double check. I see that the methods you mention are correct, can you run the demo project https://github.com/hyperoslo/ImagePicker/tree/master/Demo/ImagePickerDemo ?

**Comment by pjr0009** [comment]
@onmyway133 i just tried that, turns out the demo project actually works for me. I've done some quick debugging and it seems that the (imagePicker: ImagePickerController) parameter is is what's causing the issue for me. If i remove that parameter from the protocol functions, it compiles just fine. Also if i replace that parameter with UIViewController it still compiles fine. However, as soon as I use imagePicke: imagePickerController as the parameter, it fails.

Also the second error I posted is able to be solved by having my view controller inherit from UINavigationControllerDelegate and UIImagePickerControllerDelegate. However I don't think that's listed in the docs? So i suspect that something is wrong here as well

fwiw I'm using swift 3.0.2

**Comment by HackShitUp** [comment]
Bump to @pjr0009's issue

**Comment by aashishdhawan** [comment]
This is strange, i am unable to reproduce this on Xcode 8.2.1 with Swift 3.0.2.
@pjr0009  @HackShitUp  are you guys still facing this issue?

**Comment by HackShitUp** [comment]
Actually got it to work with an extension, but unfortunately decided to not use this framework. Sorry :/

**Comment by ivanstang** [comment]
I also faced this issue when I was using ImagePicker module in an obj c project.
I solved this issue by following:
1. Check out ImagePicker source code
2. Open the file ImagePickerViewController.swift,  delete "@objc" before "public protocol ImagePickerDelegate: class"
3. Rebuild
4. Put the new framework file to my obj-c project
5. Rebuild my project, the issue is disappeared
I hope the workaround is useful for you.
